ui/raidboss/data/06-ew/raid/p3s.ts

```diff
@@ -320,6 +334,89 @@ const triggerSet: TriggerSet<Data> = {
         },
       },
     },
+    {
+      id: 'P3S Sunbird Tether',
+      type: 'Tether',
+      netRegex: NetRegexes.tether({ id: '0039' }),
+      promise: async (data) => {
+        const callData = await callOverlayHandler({
+          call: 'getCombatants',
+        });
+        if (!callData || !callData.combatants || !callData.combatants.length) {
+          console.error('SunbirdTether: failed to get combatants: ${JSON.stringify(callData)}');
+          return;
+        }
+        const sunbirds = callData.combatants.filter((c) => c.BNpcID === 13635);
+        if (sunbirds.length !== 4) {
+          console.error('SunbirdTether: There is not exactly four Sunbirds?!?: ${JSON.stringify(sunbirds)}');
+          return;
+        }
+        data.sunbirds = sunbirds;
+      },
```

Comment:
Does this need to run on every regex match?

Reply:
No idea if the situation changes because getCombatants can't be emulated correctly by the raidemulator. I definitely need it before first tether to populate the list.

Also I don't know of a way to limit the promise execution with a condition. Is that even possible? Best I could do is immediately resolve the promise if it's the wrong trigger.

Reply:
You could accumulate all of the tethers in a separate trigger, and then add a `condition` here.  I think it depends on when you can get the information for the birds.  Ideally you get the bird information as soon as possible (maybe as soon as they become targetable, maybe there's an added combatant line and you don't need get combatants?), and then when tethers appear many seconds later, you can make calls from there.

Raidemulator should be able to handle getCombatants here, imo.  The birds are around and have plenty of data about where they are while people are hitting them.

Reply:
Depending on how tether retargeting works I have to refresh the sunbird's position when it tethers. I think if the bird has no more tethers but a charge left it just picks a new tether target as happened here: ![grafik](https://user-images.githubusercontent.com/31262/150278068-d28fad3e-6572-4ba8-8de3-e4d3e2c7a086.png)

No idea if it's actually possible to do anything about it because I remember the charges being in very rapid succession.

Reply:
> No idea if the situation changes because getCombatants can't be emulated correctly by the raidemulator. I definitely need it before first tether to populate the list.

raidemulator handles getCombatants fine for this case, for future reference. See my other version of the trigger, which works fine in raidemulator and produces the correct results according to VOD checking.

Reply:
No offense meant against the raidemulator I just don't know how it emulates the combatant data and since it burned me on zodiark my trust in the data is low.

I _guess_ it uses all log lines to retrack combatant positions? In that case I totally agree the data is fine.